### PR TITLE
Allow usage of requestPassword attribute in LdapPasswordlessUserAcco…

### PR DESCRIPTION
**Pull request by biau**
Add support for requestPassword attribute in Ldap passwordless user account store.
Actually only simple account store support disabling passswordless webflow.

### CODE REVIEW

File: api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/passwordless/account/PasswordlessAuthenticationLdapAccountsProperties.java [hunk 1]
```diff
@@ -42,4 +42,10 @@ public class PasswordlessAuthenticationLdapAccountsProperties extends AbstractLd
      * indicates the user's name.
      */
     private String nameAttribute = "cn";
+
+    /**
+     * Name of the LDAP attribute that
```
Comment (mmoayyed) [inline api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/passwordless/account/PasswordlessAuthenticationLdapAccountsProperties.java:47]: This is incorrect. This attribute does not disable passwordless functionality. It allows the passwordless flow to request a password prompt from user.

File: support/cas-server-support-passwordless-ldap/src/test/java/org/apereo/cas/impl/account/LdapPasswordlessUserAccountStoreTests.java [hunk 2]
```diff
@@ -35,7 +35,8 @@
     "cas.authn.passwordless.accounts.ldap.bind-dn=cn=Directory Manager",
     "cas.authn.passwordless.accounts.ldap.bind-credential=password",
     "cas.authn.passwordless.accounts.ldap.email-attribute=mail",
-    "cas.authn.passwordless.accounts.ldap.phone-attribute=telephoneNumber"
+    "cas.authn.passwordless.accounts.ldap.phone-attribute=telephoneNumber",
+    "cas.authn.passwordless.accounts.ldap.request-password=false"
```
Comment (mmoayyed) [inline support/cas-server-support-passwordless-ldap/src/test/java/org/apereo/cas/impl/account/LdapPasswordlessUserAccountStoreTests.java:39]: This is invalid. There is no such setting.

File: support/cas-server-support-passwordless-ldap/src/test/java/org/apereo/cas/impl/account/LdapPasswordlessUserAccountStoreTests.java [hunk 3]
```diff
@@ -55,5 +56,6 @@ public void verifyAction() throws Exception {
         assertTrue(user.isPresent());
         assertEquals("passwordlessuser@example.org", user.get().getEmail());
         assertEquals("123456789", user.get().getPhone());
+        assertEquals(false, user.get().getRequestPassword());
```
Comment (mmoayyed) [inline support/cas-server-support-passwordless-ldap/src/test/java/org/apereo/cas/impl/account/LdapPasswordlessUserAccountStoreTests.java:59]: This test is invalid; You should design a test case that check for this flag being `true`.

**Review by mmoayyed — CHANGES_REQUESTED** [review]

File: api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/passwordless/account/PasswordlessAuthenticationLdapAccountsProperties.java (hunk 1 above)
Comment (biau) [reply api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/passwordless/account/PasswordlessAuthenticationLdapAccountsProperties.java:47]: My description was based on documentation :
https://apereo.github.io/cas/development/authentication/Passwordless-Authentication.html#disabling-passwordless-authentication-flow
should I say "disable passwordless authentication flow" ?

File: support/cas-server-support-passwordless-ldap/src/test/java/org/apereo/cas/impl/account/LdapPasswordlessUserAccountStoreTests.java (hunk 2 above)
Comment (biau) [reply support/cas-server-support-passwordless-ldap/src/test/java/org/apereo/cas/impl/account/LdapPasswordlessUserAccountStoreTests.java:39]: sorry I tried I'm not a developper but I tried this with cas-overlay and it was working. What am I missing ? 
Modification of [PasswordlessAuthenticationLdapAccountsProperties.java](https://github.com/apereo/cas/pull/5600/commits/b81c899661a211cfab2c37b8a31963121250dcbb#diff-f2ff177bc35aed8a57d5658bd2c4d96e86d42a04e6dd36b551612be9d6df2670)  does not add the settings ?

Comment (mmoayyed) [reply support/cas-server-support-passwordless-ldap/src/test/java/org/apereo/cas/impl/account/LdapPasswordlessUserAccountStoreTests.java]: It does. Your setting does not match what you added. Is this setting actually called `request-password`?

File: api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/passwordless/account/PasswordlessAuthenticationLdapAccountsProperties.java (hunk 1 above)
Comment (mmoayyed) [reply api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/passwordless/account/PasswordlessAuthenticationLdapAccountsProperties.java:47]: You should say: "This setting allows the passwordless flow to request a password prompt from user."

File: support/cas-server-support-passwordless-ldap/src/test/java/org/apereo/cas/impl/account/LdapPasswordlessUserAccountStoreTests.java [hunk 4]
```diff
@@ -55,5 +56,6 @@ public void verifyAction() throws Exception {
         assertTrue(user.isPresent());
         assertEquals("passwordlessuser@example.org", user.get().getEmail());
         assertEquals("123456789", user.get().getPhone());
+        assertEquals(true, user.get().getRequestPassword());
```
Comment (mmoayyed) [inline support/cas-server-support-passwordless-ldap/src/test/java/org/apereo/cas/impl/account/LdapPasswordlessUserAccountStoreTests.java:59]: How could this test ever work? Is there a setting called `request-password`? and how could the LDAP server return true for this in this test? 

Was this actually tested?

File: support/cas-server-support-passwordless-ldap/src/test/java/org/apereo/cas/impl/account/LdapPasswordlessUserAccountStoreTests.java (hunk 2 above)
Comment (biau) [reply support/cas-server-support-passwordless-ldap/src/test/java/org/apereo/cas/impl/account/LdapPasswordlessUserAccountStoreTests.java]: ok sorry about that I completely misunderstood the way the test was working. 
Then I should load a ldap extension with an attribute requestPassword for example
Set it to true in the ldif
And map it in my config here...

Comment (mmoayyed) [reply support/cas-server-support-passwordless-ldap/src/test/java/org/apereo/cas/impl/account/LdapPasswordlessUserAccountStoreTests.java]: Yes correct, you got it.

**Pull request merged**